apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotifications.js

```diff
@@ -0,0 +1,165 @@
+
+    let count = 0
+    let sum = 0
+
+    if (!isEmpty(billingReceipts)) {
+        for (const billingReceipt of billingReceipts) {
+            const billingContext = await loadOneCached(BillingIntegrationOrganizationContext, { id: billingReceipt.context.id })
+            const serviceConsumerWhere = {
+                billingAccount: { id: billingReceipt.account.id },
+                billingIntegrationContext: { id: billingReceipt.context.id },
+                organization: { id: billingContext.organization.id },
+            }
+            const serviceConsumer = await loadOneCached(ServiceConsumer, serviceConsumerWhere)
+
+            if (!isEmpty(get(serviceConsumer, 'resident'))) {
+                const resident = await loadOneCached(Resident, { id: serviceConsumer.resident.id })
+                const locale = get(COUNTRIES, get(resident, 'residentOrganization.country')).locale || DEFAULT_LOCALE
+                const notificationKey = `${billingReceipt.period}.${billingReceipt.account.id}.${billingReceipt.category.id}`
+                const toPay = parseFloat(billingReceipt.toPay)
+                const category = await loadOneCached(BillingCategory, { id: billingReceipt.category.id })
+                const currencyCode = billingContext.integration.currencyCode || DEFAULT_CURRENCY_CODE
+
+                const data = {
+                    residentId: resident.id,
+                    userId: resident.user.id,
+                    url: `${conf.SERVER_URL}/billing/receipts/${billingReceipt.id}`,
+                    billingReceiptId: billingReceipt.id,
+                    billingAccountId: billingReceipt.account.id,
+                    billingPropertyId: billingReceipt.property.id,
+                    period: billingReceipt.period,
+                    category: category.name,
+                    toPay,
+                    currencySymbol: CURRENCY_SYMBOLS[currencyCode] || currencyCode,
+                }
+
+                const messageType = toPay > 0 ? BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE : BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE
+                const messageData = {
+                    lang: locale,
+                    to: { user: { id: resident.user.id } },
+                    type: messageType,
+                    meta: { dv: 1, data },
+                    sender: { dv: 1, fingerprint: 'sendBillingReceiptsAddedNotifications' },
+                    uniqKey: notificationKey,
+                }
+
+                if (toPay > 0) sum += toPay
+
+                if (forceSend) {
+                    try {
+                        await sendMessage(keystone, messageData)
+                    } catch (e) {
+                        // TODO(DOMA-3343): handle error type
+                    }
+
+                    await lastDtHandler.set(billingReceipt.createdAt)
+                }
+
+                count++
+            }
+
+            if (count >= maxNotificationsCount) break
```

Comment:
it look strange. You need to use the examples from @ekabardinsky

Reply:
Actually this is implemented in order to be able to send notifications portion by portion like 500 at a time. So consequent series of executions like every 10 minutes can handle any number of receipts still using limited resources.